### DISCUSSION TITLE
Unable to start a server on vercel environment

### DISCUSSION THREAD

**Discussion by ibnmhmd** [vercel/vercel#4435] [Help] [answered]
I have an eCommerce web site which’s built with the below stack

- React for client side
- Nextjs for SSR and routing
- Node and Express js for server side
- MongoDb
- Redis for caching
- IBM loopback 4 and swagger OAS for API creation

Whenever I need to run the project I execute the below two commands
npm run build-client to build the client side code and copy the built files to the desired loopback serving folder from which it serves the main project
npm start to start the server and start accepting requests.

I've deployed the above project from my **git repo**, and also provided the build **build-client** alongside the development **npm start** commands in the project setting page _**but vercel always executes the build command only without executing the development command**_ which results in running frontend only without the server being started, would you kindly advise me on the appropriate techniques/work around to handle this ? moreover, does vercel support such projects with these tech stacks ?

**Comment by styfle** [comment, marked as answer]
Hi @ibnmhmd 

Vercel is a platform for Static and Jamstack projects, typically using a Static Site Generator for your `build` script along with Serverless Functions in your `/api` directory.

Your build script should be something like `next build`.

You don't need Express because functions in the `/api` directory will have access to [express-like helpers](https://vercel.com/docs/runtimes#official-runtimes/node-js/node-js-request-and-response-objects/node-js-helpers).

In the case of Next.js, this is available using the `/pages/api` directory as seen here: https://nextjs.org/docs/api-routes/introduction

Your dev script can utilize `next dev` to run locally.

**Reply by ibnmhmd** [reply]
Thanks @styfle, I'll try to host the back-end on another platform which supports node js i.e (heroku, aws...etc) .
